Assets/scripts/PlayerController.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerController : MonoBehaviour
+{
+    [SerializeField] Animator playerAnimator;
+    [SerializeField] BoxCollider2D fullCollider;
+    public Rigidbody2D rb;
+    public float speed;
+    [SerializeField] float playerJump;
+
+    [SerializeField] LayerMask groundlayer;
+    [SerializeField] Transform groundcheckTansform;
+    public float circleRadius;
+    [SerializeField] bool isGrounded;
+   
+
+
+    void Start()
+    {
+        playerAnimator = GetComponent<Animator>();
+        fullCollider = GetComponent<BoxCollider2D>();
+        rb = GetComponent<Rigidbody2D>();       
+    }
+
+    
+    void Update()
+    {
+        isGrounded = Physics2D.OverlapCircle(groundcheckTansform.position, circleRadius, groundlayer);
+        RunPlayer();
+        PlayerCrouch();
+        PlayerJump();
+    }
+
+    private void PlayerJump()
+    {
+        if(Input.GetKeyDown(KeyCode.Space)&&isGrounded)
+        {
+            playerAnimator.SetTrigger("jump");
+            rb.velocity = new Vector2(0, playerJump);
+        }        
+    }
+
+    private void PlayerCrouch()
+    {
+
+        if (Input.GetKeyDown(KeyCode.LeftControl))
+        {
+            fullCollider.enabled = false;           
+            playerAnimator.SetBool("crouch", true);           
+        }
+        if (Input.GetKeyUp(KeyCode.LeftControl))
+        {
+            fullCollider.enabled = true;
+            playerAnimator.SetBool("crouch", false);
+        }
+
+    }
+
+    private void RunPlayer()
+    {
+        float playerXmovement = Input.GetAxisRaw("Horizontal");
```

Comment:
float playerXmovement = Input.GetAxisRaw("Horizontal");
            if (Mathf.Abs(playerXmovement) == 1)

You can save calls if you check firstly the input of the player.